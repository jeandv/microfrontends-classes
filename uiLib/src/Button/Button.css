.ui-button {
  border: 0;
  border-radius: 0.4em;
  cursor: pointer;
  display: inline-block;
  line-height: 1;
}

.ui-button--primary {
  color: #3a3a3c;
  background-color: #fc0;
}

.ui-button--primary:hover {
  background-color: #fdde47;
}

.ui-button--secondary {
  color: #333;
  background-color: transparent;
  box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset;
}

.ui-button--small {
  font-size: 12px;
  padding: 10px 16px;
}

.ui-button--medium {
  font-size: 14px;
  padding: 11px 20px;
}

.ui-button--large {
  font-size: 16px;
  padding: 12px 24px;
}