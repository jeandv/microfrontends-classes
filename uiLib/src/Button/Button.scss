.ui-button {
  background-color: #222;
  color: #fff;
  padding: 10px;
  transition: all 300ms;

  &:hover {
    background-color: #000;
  }

}

.ui-button-primary {
  font-size: 16px;
  border: 1px solid #fff;
}

.ui-button-secondary{
  font-size: 16px;
  font-weight: bold;
}