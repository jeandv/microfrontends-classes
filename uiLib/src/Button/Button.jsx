import React from 'react';
import PropTypes from 'prop-types';
import './Button.css';

const Button = ({ variant, ...props }) => {
  return (
    <button
      className={`ui-button ${variant && 'ui-button-' + variant}`}
      {...props}
    >
      Button 01
    </button>
  );
};

Button.propTypes = {
  /**
   * Is this the principal call to action on the page?
   */
  primary: PropTypes.bool,
  /**
   * What background color to use
   */
  backgroundColor: PropTypes.string,
  /**
   * How large should the button be?
   */
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  /**
   * Button contents
   */
  label: PropTypes.string.isRequired,
  /**
   * Optional click handler
   */
  onClick: PropTypes.func,
};

Button.defaultProps = {
  backgroundColor: null,
  primary: false,
  size: 'medium',
  onClick: undefined,
};

export default Button;
