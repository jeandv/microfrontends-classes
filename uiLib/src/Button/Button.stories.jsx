import React from 'react';

import Button from './Button';

export default {
  title: 'library/Button',
  component: Button,
};

const Template = (args) => <Button {...args} />;

export const Primary = Template.bind({});

// Adding Args
Primary.args = {
  primary: true,
  label: 'Comprar',
};

export const Secondary = Template.bind({});

Secondary.args = {
  primary: true,
  label: 'Comprar',
};
