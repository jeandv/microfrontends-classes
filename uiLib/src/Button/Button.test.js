import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

it('Renders correctly', () => {

  const tree = renderer
    .create(<Button label='Hello' />)
    .toJSON();

  expect(tree.props.type).toBe('button');
  expect(tree).toMatchSnapshot();

});

it('Changes when is hovered', () => {

  const component = renderer.create(<Button label='Hello' />)

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  // Checar la funcion que tiene
  renderer.act(() => {
    tree.props.onMouseEnter();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});