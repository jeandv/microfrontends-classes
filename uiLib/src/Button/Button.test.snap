// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changes when is hovered 1`] = `
<button
  backgroundColor={null}
  className="ui-button undefined"
  label="Hello"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  primary={false}
  size="medium"
>
  Hello
</button>
`;

exports[`Changes when is hovered 2`] = `
<button
  backgroundColor={null}
  className="ui-button undefined"
  label="Hello"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  primary={false}
  size="medium"
>
  Hello
</button>
`;

exports[`Renders correctly 1`] = `
<button
  backgroundColor={null}
  className="ui-button undefined"
  label="OLLLI!"
  primary={false}
  size="medium"
>
  OLLLI!
</button>
`;
